### ISSUE TITLE
Terraform Registry mitigation for go-getter download bug

### ISSUE THREAD

**Issue by dpiddockcmp** [hashicorp/terraform#23829]
### Terraform Version
All Terraform 0.12 before 0.12.19.

Terraform 0.11.14 appears to be unaffected.

### Terraform Configuration Files
```hcl
module "vpc" {
  source = "terraform-aws-modules/vpc/aws"
}
```

### Expected Behavior
Module gets downloaded

### Actual Behavior
```
$ terraform init
Initializing modules...
Downloading terraform-aws-modules/vpc/aws 2.21.0 for vpc...

Error: Failed to download module

Could not download module "vpc" (main.tf:1) source code from
"https://api.github.com/repos/terraform-aws-modules/terraform-aws-vpc/tarball/v2.21.0//*?archive=tar.gz":
Error opening a gzip reader for
/var/folders/mv/zvbyy31x0l33852hj2xws73c0000gn/T/getter639019060/archive: EOF.


Error: Failed to download module

Could not download module "vpc" (main.tf:1) source code from
"https://api.github.com/repos/terraform-aws-modules/terraform-aws-vpc/tarball/v2.21.0//*?archive=tar.gz":
Error opening a gzip reader for
/var/folders/mv/zvbyy31x0l33852hj2xws73c0000gn/T/getter639019060/archive: EOF.
```

### Steps to Reproduce
1. `rm -r .terraform`
1. `terraform init`

### Additional Context
We really can't go through all our terraform configs and CI setups at the drop of a hat and update to the latest version.

### References
- #23824
- https://github.com/terraform-aws-modules/terraform-aws-eks/issues/676
- https://github.com/terraform-aws-modules/terraform-aws-eks/issues/672
- https://github.com/terraform-aws-modules/terraform-aws-eks/issues/674
- https://github.com/terraform-aws-modules/terraform-aws-vpc/issues/380
- https://github.com/terraform-aws-modules/terraform-aws-rds/issues/185

**Comment by seboss666** [comment]
I can confirm this, we tried the same 0.12.18 version on the same project we lastly modified before christmas, and it's working. 0.12.19 does not mention any change on URL handling for module download.

We confirm the 0.12.19 "fix" on at least two computers. Yesterday the 0.12.18 was still working correctly. Could it be a change in github api behaviour or something ?

**Comment by luckynoodle** [comment]
We had to change the source to source = "github.com/terraform-aws-modules/terraform-aws-vpc?ref=v2.21.0" to make it work

**Comment by jbardin** [comment]
Hi @dpiddockcmp,

This issue is recorded in #23721. While the terraform bug is fixed in 0.12.19, the registry team is working on a mitigation to redirect the module requests to a URL that is unlikely to trigger the bug in older versions.

Thanks

**Comment by dpiddockcmp** [comment]
Is there an open Issue somewhere for tracking where the registry team is working on a fix? This has been broken almost a month! That's pretty disappointing.

**Comment by HaithemSouala** [comment]
same issue with **terraform-google-modules/vault/google** module

**Comment by jbardin** [comment]
Unfortunately a recent change in GH infrastructure is what triggered the bug in a Terraform dependency. The only way to update the dependency is to release a new Terraform version, which in this case is 0.12.19. Earlier 0.12 versions will always contain this bug, and the best way to avoid the problem is to upgrade. 

Since the Terraform issue is fixed, I can change this issue to track any registry changes, and we can close it once a determination is made if a workaround is possible at all for older versions.

**Comment by jbardin** [comment]
Closed via #23834

**Comment by dpiddockcmp** [comment]
Awesome. Thank you for the updates 😄
